service/kernel/data_management/finiteness_checker.cpp

```diff
@@ -129,6 +140,7 @@ DataType computeSumAVX512Impl(size_t nDataPtrs, size_t nElementsPerPtr, const Da
 
     daal::services::internal::TArray<DataType, avx512> partialSumsArr(nTotalBlocks);
     DataType * pSums = partialSumsArr.get();
+    if (!pSums) return getInf<DataType>();
     for (size_t iBlock = 0; iBlock < nTotalBlocks; ++iBlock) pSums[iBlock] = 0;
 
     runBlocks(inParallel, nTotalBlocks, [&](size_t iBlock) {
```

Comment:
I'd prefer to follow code guideline along this file
http://intel.github.io/daal/contribution/coding_guide.html#usage-of-const